Source/ProgressionSystemRuntime/Public/Components/PSHUDComponent.h

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#pragma once
+
+#include "Data/PSTypes.h"
+#include "Data/PSDataAsset.h"
+#include "Components/ActorComponent.h"
+#include "Structures/PlayerTag.h"
+//---
+#include "PSHUDComponent.generated.h"
+
+/**
+ * Implements the core logic on project about Progression System.
+ */
+
+UCLASS(Blueprintable, ClassGroup=(Custom), meta=(BlueprintSpawnableComponent))
+class PROGRESSIONSYSTEMRUNTIME_API UPSHUDComponent final : public UActorComponent
+{
+	GENERATED_BODY()
+
+public:
+	/** Sets default values for this component's properties. */
+	UPSHUDComponent();
+
+	/** Returns the Progression System data asset. */
+	UFUNCTION(BlueprintPure, Category = "C++")
+	static const UPSDataAsset* GetProgressionSystemDataAsset() { return &UPSDataAsset::Get(); }
+
+	/** Returns current saved progression. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	const FORCEINLINE FPSRowData& GetSavedProgressionRowData() const { return SavedProgressionRowDataInternal; }
+
+	/** Save the progression depends on EEndGameState. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	void SavePoints(ELevelType Map, FPlayerTag Character, EEndGameState EndGameState);
+
+	/** Returns the endgame reward. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	int32 GetProgressionReward(ELevelType Map, FPlayerTag Character, EEndGameState EndGameState);
+
+	/** Saves the current progression. */
+	UFUNCTION(BlueprintCallable, Category="C++")
+	void SaveCurrentGameProgression();
+
+	UFUNCTION()
+	void SaveDataAsync();
```

Comment:
- What is the purpose of this function? Please explain in code comment above UFUNCTION.
- Should it be in Public?

Reply:
SavePoints performs some data changes so save file needs to be updated. I've improved this, removed local save at component level, used global from world sub system.